File: C#_DesignPattern/Proxy/Proxy/Proxy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Proxy
{
    public class Proxy : GiveGift
    {
        Pursuit gg;
        public Proxy(SchoolGirl mm)
        {
            this.gg = new Pursuit(mm);
        }

        public void GiveDolls()
        {
            gg.GiveDolls();
        }

        public void GiveFlowers()
        {
            gg.GiveFlowers();
        }

        public void GiveChocolate()
        {
            gg.GiveChocolate();
        }

    }
}

File: C++_DesignPattern/Proxy/Proxy/Proxy.cpp
// Proxy.cpp : 定义控制台应用程序的入口点。
//

#include "stdafx.h"
#include "Givegift.h"

int _tmain(int argc, _TCHAR* argv[])
{
	SchoolGirl *jiaojiao = new SchoolGirl();
	jiaojiao->name = "李娇娇";

	Proxy *daili = new Proxy(jiaojiao);

	daili->GiveDolls();
	daili->GiveFlowers();
	daili->GiveChocolate();
	return 0;
}

